### ISSUE TITLE
NIRSpec ASN product names need multiple optical elements

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#220]
CV3 SIC-DIL program 93135 generated some NIRSpec ASN's, but the product names in the ASN tables only included the filter name in the optical-element field. They should include both filter and grating names, as is done with NIRCam and NIRISS filter and pupil names.

For example, in jw93135_20160908t174455_spec_286_asn.json:

```
"name": "jw93135-a3001_t001_nirspec_f170lp_{product_type}.fits",

"constraints": "Constraints:\n ... opt_elem: F170LP\n  opt_elem2: NULL\n ... grating: G235H
```

**Comment by stscieisenhamer** [comment]
# Implementation
- [ ] Dependent on resolution to #238 
- [ ] Dependent on resolution to #240

**Comment by stscieisenhamer** [comment]
Obi-Wan's failure is now complete: This issue is relevant to all NIRSpec modes except IFU. For IFU, only grating will be used; not filter at all. This issue will be closed when fixed for all modes; #238 will track the IFU special case.
